server/config/authUtil.js

```diff
@@ -0,0 +1,80 @@
+var bcrypt = require('bcrypt-nodejs');
+var User = require('../users/userModel.js');
+var jwt  = require('jwt-simple');
+var tokenKey = 'u8yUOyiBjLXJqj9opqnKjhWoc0pm8fyl';
+
+module.exports = {
+
+  isTokenInHeader: function(req){
+    return !!req.headers['x-access-token'];
+  },
+
+  encodeToken: function(payload){
+    return jwt.encode(payload, tokenKey);
+  },
+
+  decodeToken: function(token){
+    return jwt.decode(token, tokenKey);
+  },
+
+  decodeTokenAttachToReq: function(req, res, cbNext){
+    var token = req.headers['x-access-token'];
+    if (token) {
+      req.decodedToken = jwt.decode(token, tokenKey);
+      cbNext();
+    } else {
+      res.sendStatus(403);
+    }
+  },
+
+  validatePassword: function(reqUser, dbUser, onFail, onSuccess){
+    bcrypt.compare(reqUser.password, dbUser.password, function(err, passwordsMatch){
+
+      if (err) console.error(err);
+      if (passwordsMatch) {
+        var token = jwt.encode({ username: reqUser.username }, tokenKey);
```

Comment:
Check with @onelovelyname on what we should encode for the token. If it's just username, I think we can simply encode `username` instead of an object with username as the only key-value pair.